### PR TITLE
Fix(inventory, stock-location): Remove orphaned location levels and reservations

### PR DESCRIPTION
**Pull request by pKorsholm**
**What**
- Remove related inventory levels and reservation items when a stock location is removed

**How**
- Add bulk deletion methods for both inventory levels and reservation items to the inventory service api
- invoke both on location removal

Fixes CORE-1232

### CODE REVIEW

File: packages/inventory/src/services/inventory-level.ts [hunk 1]
```diff
@@ -21,6 +21,7 @@ export default class InventoryLevelService extends TransactionBaseService {
     CREATED: "inventory-level.created",
     UPDATED: "inventory-level.updated",
     DELETED: "inventory-level.deleted",
+    DELETED_BY_LOCATION: "inventory-level.deleted-by-location",
```
Comment (srindom) [inline packages/inventory/src/services/inventory-level.ts:24]: suggestion: rename to BULK_DELETED and allow the event data to be dynamic based on the query. I.e.: Event data could be: `{ location_id: ... }` but `{ inventory_item_id: ... }` would also be valid. 

This way we don't have to create `DELETED_BY_X` for every deletion dimension. People who want to build synchronization could simply listen to `DELETED` and `BULK_DELETED` events.

@adrien2p, @olivermrbl - curious to hear your thoughts on this too?

File: packages/inventory/src/services/reservation-item.ts [hunk 2]
```diff
@@ -24,6 +24,7 @@ export default class ReservationItemService extends TransactionBaseService {
     CREATED: "reservation-item.created",
     UPDATED: "reservation-item.updated",
     DELETED: "reservation-item.deleted",
+    DELETED_BY_LOCATION: "reservation-item.deleted-by-location",
     DELETED_BY_LINE_ITEM: "reservation-item.deleted-by-line-item",
```
Comment (srindom) [inline packages/inventory/src/services/reservation-item.ts:28]: suggestion: same as for levels.

File: packages/inventory/src/services/inventory-level.ts (hunk 1 above)
Comment (srindom) [reply packages/inventory/src/services/inventory-level.ts:24]: Could you give an example maybe? :)

Comment (srindom) [reply packages/inventory/src/services/inventory-level.ts:24]: I would be fine with that too -- would just have to be documented.

Comment (adrien2p) [reply packages/inventory/src/services/inventory-level.ts:24]: Eventually, second options is 

```ts
const { raw } = levelRepository.delete({ location_id: locationId })
const events = raw.map(level => ({ eventName: InventoryLevel.Events.DELETED, data: { id: level.id })
await this.eventBusService_.withTransaction(manager).emit(events)
```

Wdyt @srindom ?

Comment (pKorsholm) [reply packages/inventory/src/services/inventory-level.ts]: Going for the pure deleted event, where do you want it documented @srindom?

Comment (srindom) [reply packages/inventory/src/services/inventory-level.ts]: cc: @shahednasser, can we document this behavior in the events reference :)

**Review by srindom — COMMENTED** [review]
Please add a test to ensure that this logic works as correctly :)

File: packages/inventory/src/services/inventory-level.ts (hunk 1 above)
Comment (shahednasser) [reply packages/inventory/src/services/inventory-level.ts]: @srindom sure, once it's merged I'll include this behavior in the events reference (creating a linear issue for it) 👍🏻

File: packages/admin-ui/ui/src/domain/inventory/locations/components/location-card/index.tsx [hunk 3]
```diff
@@ -37,7 +37,7 @@ const LocationCard: React.FC<Props> = ({ location }) => {
   const onDelete = async () => {
     const shouldDelete = await dialog({
       heading: "Delete Location",
-      text: "Are you sure you want to delete this location",
+      text: "Are you sure you want to delete this location. This will also delete all inventory levels and reservations associated with this location.",
```
Comment (srindom) [inline packages/admin-ui/ui/src/domain/inventory/locations/components/location-card/index.tsx:40]: praise: great UX addition!

**Comment by srindom** [comment]
@pKorsholm - can you check up on the failing test :)

**Comment by srindom** [comment]
The integration problem feels like a missing await - can't find any though; could it be because we don't do `return await` in delete functions?

**Comment by adrien2p** [comment]
> The integration problem feels like a missing await - can't find any though; could it be because we don't do `return await` in delete functions?

on Master I had the same issue and it was due to missing transaction during the emit, might be worth looking at the diff with master

**Comment by pKorsholm** [comment]
Clutch @adrien2p! Thanks, that was the issue 😅

**Pull request merged**